src/plot_api/plot_api.js

```diff
@@ -2539,7 +2539,7 @@ function getTraceIndexFromUid(uid, data, tracei) {
         if(data[i].uid === uid) return i;
     }
     // fall back on trace order, but only if user didn't provide a uid for that trace
-    return data[tracei].uid ? -1 : tracei;
+    return (!data[tracei] || data[tracei].uid) ? -1 : tracei;
```

Comment:
Wondering if we may add few more checks to test `tracei` to be a valid index?
<pre>
return (tracei === undefined || tracei < 0 || !data[tracei] || data[tracei].uid) ? -1 : tracei;
</pre>

Reply:
`getTraceIndexFromUid` gets called here:

https://github.com/plotly/plotly.js/blob/7d117741d0e93f79831b65faa78a113bf3d8109b/src/plot_api/plot_api.js#L2628

and `index` is filled here:

https://github.com/plotly/plotly.js/blob/7d117741d0e93f79831b65faa78a113bf3d8109b/src/plots/plots.js#L1019

so I can't be anything other than positive number.